src/ape/cli/commands.py

```diff
@@ -1,26 +1,43 @@
 import inspect
 import warnings
-from typing import Any, List
+from typing import Any, List, Optional
 
 import click
 from click import Context
 
 from ape import networks
-from ape.api import ProviderAPI
+from ape.api import ProviderAPI, ProviderContextManager
 from ape.cli.choices import NetworkChoice
 from ape.exceptions import NetworkError
 
 
-def check_parents_for_interactive(ctx: Context) -> bool:
-    interactive: bool = ctx.params.get("interactive", False)
-    if interactive:
-        return True
+def get_param_from_ctx(ctx: Context, param: str) -> Optional[Any]:
+    if value := ctx.params.get(param):
+        return value
 
     # If not found, check the parent context.
-    if interactive is None and ctx.parent:
-        return check_parents_for_interactive(ctx.parent)
-
-    return False
+    elif parent := ctx.parent:
+        return get_param_from_ctx(parent, param)
+
+    return None
+
+
+def parse_network(ctx: Context) -> ProviderContextManager:
+    interactive = get_param_from_ctx(ctx, "interactive")
+    provider = get_param_from_ctx(ctx, "network")
+    if provider is not None and isinstance(provider, ProviderAPI):
+        return provider.network.use_provider(provider, disconnect_on_exit=not interactive)
```

Comment:
bug2: ` disconnect_on_exit=not ` was only on one of the cases here, it didnt occur on the `elif` statements below... so that setting didn't work